molgenis-data/src/main/java/org/molgenis/data/meta/model/EntityMetaData.java

```diff
@@ -443,28 +442,28 @@ public EntityMetaData setExtends(EntityMetaData extends_)
 	 * @return A {@link List} of {@link AttributeMetaData} containing all own attributes, with compound attributes being placed after
 	 * their respective attribute parts
 	 */
-	public Set<AttributeMetaData> getCompoundOrderedAttributes()
+	public List<AttributeMetaData> getCompoundOrderedAttributes()
 	{
-		Set<AttributeMetaData> attributes = newHashSet();
+		List<AttributeMetaData> attributes = newArrayList();
 		getEntities(ATTRIBUTES, AttributeMetaData.class).forEach(attribute -> {
 			if (attribute.getDataType() == COMPOUND)
 			{
 				attribute.getAttributeParts()
 						.forEach(attributePart -> resolvePossibleNestedCompounds(attributePart, attributes));
 			}
-			attributes.add(attribute);
+			if (!attributes.contains(attribute)) attributes.add(attribute);
```

Comment:
1) Potentially expensive check for large lists. Use a Set or a LinkedSet in case you want to preserve order. 2) Coding standard: add curly braces for one line if statements